iphone/Maps/Classes/CarPlay/CarPlayService.swift

```diff
@@ -60,6 +60,33 @@ final class CarPlayService: NSObject {
     FrameworkHelper.updatePositionArrowOffset(false, offset: 5)
   }
 
+  private var savedInterfaceController: CPInterfaceController?
```

Comment:
I'm not very happy about this field. We already have the `interfaceController` in this class, so maybe we can preserve it during the switch to the phone by writing a separate `switch()` method instead of calling `destroy`?

Reply:
The idea was the following: we want to behave the same way as if user disconnected phone from car. Keeping `interfaceController` non null can have side effects.

For example I see `[[MWMCarPlayService shared] showNoMapAlert]` call from `MWMMapDownloadDialog`, which shows alert if `interfaceController` is not null. I did not test if there is real problems in that case, but we have 26 references to `interfaceController` which I prefer not to think about.

By using separate variable we can limit usage of `interfaceController` to exaclty 2 things:
1. display "phone mode" alert
2. use it to recover carplay mode when user asks for this

Howewer we should probably nullify `savedInterfaceController` on real CarPlay disconnect event, so separated switch() and destroy() methods will be useful.

Reply:
Ok I see your point but I don't think we want the same behavior as if the user disconnects the phone from the car because the phone is still connected. We just want to switch the screen from CarPlay to the phone. So maybe you just want to call the code that switches the screen without clearing all the references that we might need later.

Anyway, I understand if you don't want to spend to much time on this PR and the switching works so that might be a refactoring for later.

Reply:
I created 2 functions `switchScreenToPhone()` and `destroy()` to distinguish switch and destroy, it will be easier to move logic in the future if we decide to do it.

Reply:
Ok looks good to me :)